### ISSUE TITLE
Field tl_(news|calendar_events).pid does not seem to be related

### ISSUE THREAD

**Issue by bennyborn** [contao/contao#5014]
### Affected version(s)

5.0.x-dev, 5.0

### Description

When trying to edit a news article in the backend the following exception occurs:

```
Field tl_news.pid does not seem to be related

  at vendor/contao/core-bundle/contao/library/Contao/Model.php:651
  at Contao\Model->getRelated()
     (vendor/contao/news-bundle/contao/classes/News.php:351)
  at Contao\News::generateNewsUrl()
     (vendor/contao/news-bundle/contao/dca/tl_news.php:657)
  at tl_news->getSerpUrl()
     (vendor/contao/core-bundle/contao/widgets/SerpPreview.php:159)
  at Contao\SerpPreview->getUrl()
     (vendor/contao/core-bundle/contao/widgets/SerpPreview.php:51)
  at Contao\SerpPreview->generate()
     (vendor/contao/core-bundle/contao/library/Contao/Widget.php:638)
  at Contao\Widget->generateWithError()
     (vendor/contao/core-bundle/contao/templates/backend/be_widget.html5:3)
  at include('/home/benny/www/contao5_php8/vendor/contao/core-bundle/contao/templates/backend/be_widget.html5')
     (vendor/contao/core-bundle/contao/library/Contao/TemplateInheritance.php:109)
  at Contao\Widget->inherit()
     (vendor/contao/core-bundle/contao/library/Contao/Widget.php:585)
  at Contao\Widget->parse()
     (vendor/contao/core-bundle/contao/classes/DataContainer.php:729)
  at Contao\DataContainer->row()
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:2014)
  at Contao\DC_Table->edit()
     (vendor/contao/core-bundle/contao/classes/Backend.php:544)
  at Contao\Backend->getBackendModule()
     (vendor/contao/core-bundle/contao/controllers/BackendMain.php:149)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:44)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:152)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw()
     (vendor/symfony/http-kernel/HttpKernel.php:74)
  at Symfony\Component\HttpKernel\HttpKernel->handle()
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle()
     (public/index.php:44)    
```

and the same happens for the events:

```
Field tl_calendar_events.pid does not seem to be related

  at vendor/contao/core-bundle/contao/library/Contao/Model.php:651
  at Contao\Model->getRelated()
     (vendor/contao/calendar-bundle/contao/classes/Events.php:461)
  at Contao\Events::generateEventUrl()
     (vendor/contao/calendar-bundle/contao/dca/tl_calendar_events.php:725)
  at tl_calendar_events->getSerpUrl()
     (vendor/contao/core-bundle/contao/widgets/SerpPreview.php:159)
  at Contao\SerpPreview->getUrl()
     (vendor/contao/core-bundle/contao/widgets/SerpPreview.php:51)
  at Contao\SerpPreview->generate()
     (vendor/contao/core-bundle/contao/library/Contao/Widget.php:638)
  at Contao\Widget->generateWithError()
     (vendor/contao/core-bundle/contao/templates/backend/be_widget.html5:3)
  at include('/home/benny/www/contao5_php8/vendor/contao/core-bundle/contao/templates/backend/be_widget.html5')
     (vendor/contao/core-bundle/contao/library/Contao/TemplateInheritance.php:109)
  at Contao\Widget->inherit()
     (vendor/contao/core-bundle/contao/library/Contao/Widget.php:585)
  at Contao\Widget->parse()
     (vendor/contao/core-bundle/contao/classes/DataContainer.php:729)
  at Contao\DataContainer->row()
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:2014)
  at Contao\DC_Table->edit()
     (vendor/contao/core-bundle/contao/classes/Backend.php:544)
  at Contao\Backend->getBackendModule()
     (vendor/contao/core-bundle/contao/controllers/BackendMain.php:149)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:44)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:152)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw()
     (vendor/symfony/http-kernel/HttpKernel.php:74)
  at Symfony\Component\HttpKernel\HttpKernel->handle()
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle()
     (public/index.php:44)                
```

**Comment by ausi** [comment]
This seems to be caused by a loop (if the DCA cache is empty), my stacktrace:

```
#0 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(112): Contao\DcaExtractor->createExtract()
#1 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(134): Contao\DcaExtractor->__construct('tl_article')
#2 …/vendor/contao/contao/core-bundle/contao/library/Contao/Database/Installer.php(188): Contao\DcaExtractor::getInstance('tl_article')
#3 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php(301): Contao\Database\Installer->getFromDca()
#4 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php(40): Contao\CoreBundle\Doctrine\Schema\DcaSchemaProvider->getSqlDefinitions()
#5 …/vendor/contao/contao/core-bundle/src/EventListener/DoctrineSchemaListener.php(32): Contao\CoreBundle\Doctrine\Schema\DcaSchemaProvider->appendToSchema(Object(Doctrine\DBAL\Schema\Schema))
#6 …/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php(66): Contao\CoreBundle\EventListener\DoctrineSchemaListener->postGenerateSchema(Object(Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs))
#7 …/vendor/doctrine/orm/lib/Doctrine/ORM/Tools/SchemaTool.php(408): Symfony\Bridge\Doctrine\ContainerAwareEventManager->dispatchEvent('postGenerateSch...', Object(Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs))
#8 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/SchemaProvider.php(34): Doctrine\ORM\Tools\SchemaTool->getSchemaFromMetadata(Array)
#9 …/vendor/contao/contao/core-bundle/contao/library/Contao/Model.php(397): Contao\CoreBundle\Doctrine\Schema\SchemaProvider->createSchema()
#10 …/vendor/contao/contao/core-bundle/contao/library/Contao/Model.php(443): Contao\Model::getColumnCastTypes()
#11 …/vendor/contao/contao/core-bundle/contao/library/Contao/User.php(271): Contao\Model::convertToPhpValue('id', 1)
#12 …/vendor/contao/contao/core-bundle/contao/library/Contao/User.php(344): Contao\User->findBy('username', 'admin')
#13 …/vendor/contao/contao/core-bundle/contao/classes/BackendUser.php(104): Contao\User::loadUserByIdentifier('admin')
#14 …/vendor/contao/contao/core-bundle/contao/dca/tl_article.php(198): Contao\BackendUser::getInstance()
#15 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaLoader.php(117): include('/Users/martin/S...')
#16 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaLoader.php(82): Contao\DcaLoader->loadDcaFiles()
#17 …/vendor/contao/contao/core-bundle/contao/library/Contao/Controller.php(1175): Contao\DcaLoader->load()
#18 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(344): Contao\Controller::loadDataContainer('tl_article')
#19 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(112): Contao\DcaExtractor->createExtract()
#20 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(134): Contao\DcaExtractor->__construct('tl_article')
#21 …/vendor/contao/contao/core-bundle/contao/library/Contao/Database/Installer.php(188): Contao\DcaExtractor::getInstance('tl_article')
#22 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php(301): Contao\Database\Installer->getFromDca()
#23 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php(40): Contao\CoreBundle\Doctrine\Schema\DcaSchemaProvider->getSqlDefinitions()
#24 …/vendor/contao/contao/core-bundle/src/EventListener/DoctrineSchemaListener.php(32): Contao\CoreBundle\Doctrine\Schema\DcaSchemaProvider->appendToSchema(Object(Doctrine\DBAL\Schema\Schema))
#25 …/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php(66): Contao\CoreBundle\EventListener\DoctrineSchemaListener->postGenerateSchema(Object(Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs))
#26 …/vendor/doctrine/orm/lib/Doctrine/ORM/Tools/SchemaTool.php(408): Symfony\Bridge\Doctrine\ContainerAwareEventManager->dispatchEvent('postGenerateSch...', Object(Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs))
#27 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/SchemaProvider.php(34): Doctrine\ORM\Tools\SchemaTool->getSchemaFromMetadata(Array)
#28 …/vendor/contao/contao/core-bundle/contao/library/Contao/Model.php(397): Contao\CoreBundle\Doctrine\Schema\SchemaProvider->createSchema()
#29 …/vendor/contao/contao/core-bundle/contao/library/Contao/Model.php(443): Contao\Model::getColumnCastTypes()
#30 …/vendor/contao/contao/core-bundle/contao/library/Contao/User.php(271): Contao\Model::convertToPhpValue('id', 1)
#31 …/vendor/contao/contao/core-bundle/contao/library/Contao/User.php(344): Contao\User->findBy('username', 'admin')
#32 …/vendor/contao/contao/core-bundle/contao/classes/BackendUser.php(104): Contao\User::loadUserByIdentifier('admin')
#33 …/vendor/contao/contao/news-bundle/contao/dca/tl_news.php(173): Contao\BackendUser::getInstance()
#34 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaLoader.php(117): include('/Users/martin/S...')
#35 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaLoader.php(82): Contao\DcaLoader->loadDcaFiles()
#36 …/vendor/contao/contao/core-bundle/contao/library/Contao/Controller.php(1175): Contao\DcaLoader->load()
#37 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaLoader.php(211): Contao\Controller::loadDataContainer('tl_news')
#38 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaLoader.php(122): Contao\DcaLoader->setDynamicPTable()
#39 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaLoader.php(82): Contao\DcaLoader->loadDcaFiles()
#40 …/vendor/contao/contao/core-bundle/contao/library/Contao/Controller.php(1175): Contao\DcaLoader->load()
#41 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(344): Contao\Controller::loadDataContainer('tl_content')
#42 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(112): Contao\DcaExtractor->createExtract()
#43 …/vendor/contao/contao/core-bundle/contao/library/Contao/DcaExtractor.php(134): Contao\DcaExtractor->__construct('tl_content')
#44 …/vendor/contao/contao/core-bundle/contao/library/Contao/Database/Installer.php(188): Contao\DcaExtractor::getInstance('tl_content')
#45 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php(301): Contao\Database\Installer->getFromDca()
#46 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php(40): Contao\CoreBundle\Doctrine\Schema\DcaSchemaProvider->getSqlDefinitions()
#47 …/vendor/contao/contao/core-bundle/src/EventListener/DoctrineSchemaListener.php(32): Contao\CoreBundle\Doctrine\Schema\DcaSchemaProvider->appendToSchema(Object(Doctrine\DBAL\Schema\Schema))
#48 …/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php(66): Contao\CoreBundle\EventListener\DoctrineSchemaListener->postGenerateSchema(Object(Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs))
#49 …/vendor/doctrine/orm/lib/Doctrine/ORM/Tools/SchemaTool.php(408): Symfony\Bridge\Doctrine\ContainerAwareEventManager->dispatchEvent('postGenerateSch...', Object(Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs))
#50 …/vendor/contao/contao/core-bundle/src/Doctrine/Schema/SchemaProvider.php(34): Doctrine\ORM\Tools\SchemaTool->getSchemaFromMetadata(Array)
#51 …/vendor/contao/contao/core-bundle/contao/library/Contao/Model.php(397): Contao\CoreBundle\Doctrine\Schema\SchemaProvider->createSchema()
#52 …/vendor/contao/contao/core-bundle/contao/library/Contao/Model.php(443): Contao\Model::getColumnCastTypes()
#53 …/vendor/contao/contao/core-bundle/contao/library/Contao/User.php(271): Contao\Model::convertToPhpValue('id', 1)
#54 …/vendor/contao/contao/core-bundle/contao/library/Contao/User.php(344): Contao\User->findBy('username', 'admin')
#55 …/vendor/contao/contao/core-bundle/src/Framework/Adapter.php(40): Contao\User::loadUserByIdentifier('admin')
#56 …/vendor/contao/contao/core-bundle/src/Security/User/ContaoUserProvider.php(46): Contao\CoreBundle\Framework\Adapter->__call('loadUserByIdent...', Array)
#57 …/vendor/contao/contao/core-bundle/src/Security/User/ContaoUserProvider.php(61): Contao\CoreBundle\Security\User\ContaoUserProvider->loadUserByIdentifier('admin')
#58 …/vendor/symfony/security-http/Firewall/ContextListener.php(222): Contao\CoreBundle\Security\User\ContaoUserProvider->refreshUser(Object(Contao\BackendUser))
#59 …/vendor/symfony/security-http/Firewall/ContextListener.php(132): Symfony\Component\Security\Http\Firewall\ContextListener->refreshUser(Object(Symfony\Component\Security\Http\Authenticator\Token\PostAuthenticationToken))
#60 …/vendor/symfony/security-bundle/Debug/WrappedLazyListener.php(49): Symfony\Component\Security\Http\Firewall\ContextListener->authenticate(Object(Symfony\Component\HttpKernel\Event\RequestEvent))
#61 …/vendor/symfony/security-http/Firewall/AbstractListener.php(26): Symfony\Bundle\SecurityBundle\Debug\WrappedLazyListener->authenticate(Object(Symfony\Component\HttpKernel\Event\RequestEvent))
#62 …/vendor/symfony/security-bundle/Debug/TraceableFirewallListener.php(73): Symfony\Component\Security\Http\Firewall\AbstractListener->__invoke(Object(Symfony\Component\HttpKernel\Event\RequestEvent))
#63 …/vendor/symfony/security-http/Firewall.php(92): Symfony\Bundle\SecurityBundle\Debug\TraceableFirewallListener->callListeners(Object(Symfony\Component\HttpKernel\Event\RequestEvent), Object(Generator))
#64 …/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(115): Symfony\Component\Security\Http\Firewall->onKernelRequest(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
#65 …/vendor/symfony/event-dispatcher/EventDispatcher.php(230): Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
#66 …/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\Component\EventDispatcher\EventDispatcher->callListeners(Array, 'kernel.request', Object(Symfony\Component\HttpKernel\Event\RequestEvent))
#67 …/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(153): Symfony\Component\EventDispatcher\EventDispatcher->dispatch(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request')
#68 …/vendor/symfony/http-kernel/HttpKernel.php(128): Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request')
#69 …/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
#70 …/vendor/symfony/http-kernel/Kernel.php(202): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#71 …/public/index.php(44): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request))
```

**Comment by ausi** [comment]
Closed in favor of #5147
